frappe/database/mariadb/schema.py

```diff
@@ -29,9 +33,27 @@ def create(self):
 				)
 			) + ',\n'
 
+		# creating sequence(s)
+		if (not self.meta.issingle \
+			and self.meta.autoname == "autoincrement") or self.doctype in log_types:
+
+			# NOTE: using a very small cache - as during backup, if the sequence was used in anyform,
+			# it drops the cache and uses the next non cached value in setval func and
+			# puts that in the backup file, which will start the counter
+			# from that value when inserting any new record in the doctype.
+			# By default the cache is 1000 which will mess up the sequence when
+			# using the system after a restore.
+			# issue link: https://jira.mariadb.org/browse/MDEV-21786
+			create_sequence(self.doctype, check_not_exists=True, cache=50)
```

Comment:
So I realized that serial has a major issue, it will become locking in case of parallel processing - specially for log type tables - I think we need a non-locking system so auto increment might work better? 😅

Reply:
The serial records that postgres or mariadb creates for each sequence is non locking.